omx/gstomxjpegenc.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2013 RidgerRun LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation
+ * version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <gst/gst.h>
+
+#include <OMX_Component.h>
+#include <OMX_TI_Index.h>
+#include <OMX_TI_Video.h>
+
+#include "gstomxjpegenc.h"
+
+GST_DEBUG_CATEGORY_STATIC (gst_omx_jpeg_enc_debug_category);
+#define GST_CAT_DEFAULT gst_omx_jpeg_enc_debug_category
+
+/* prototypes */
+static void gst_omx_jpeg_enc_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec);
+static void gst_omx_jpeg_enc_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec);
+
+static gboolean gst_omx_jpeg_enc_set_format (GstOMXVideoEnc * enc,
+    GstOMXPort * port, GstVideoCodecState * state);
+static GstCaps *gst_omx_jpeg_enc_get_caps (GstOMXVideoEnc * enc,
+    GstOMXPort * port, GstVideoCodecState * state);
+
+enum
+{
+  PROP_0,
+  PROP_QUALITY
+};
+
+#define GST_OMX_JPEG_ENC_QUALITY_DEFAULT 90
+
+/* class initialization */
+
+#define DEBUG_INIT \
+  GST_DEBUG_CATEGORY_INIT (gst_omx_jpeg_enc_debug_category, "omxjpegenc", 0, \
+      "OMX JPEG encoder element");
+
+G_DEFINE_TYPE_WITH_CODE (GstOMXJpegEnc, gst_omx_jpeg_enc,
+    GST_TYPE_OMX_VIDEO_ENC, DEBUG_INIT);
+
+static void
+gst_omx_jpeg_enc_class_init (GstOMXJpegEncClass * klass)
+{
+  GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
+  GstElementClass *element_class = GST_ELEMENT_CLASS (klass);
+  GstOMXVideoEncClass *videoenc_class = GST_OMX_VIDEO_ENC_CLASS (klass);
+
+  gobject_class->set_property = gst_omx_jpeg_enc_set_property;
+  gobject_class->get_property = gst_omx_jpeg_enc_get_property;
+
+  g_object_class_install_property (gobject_class, PROP_QUALITY,
+      g_param_spec_int ("quality", "MJPEG/JPEG quality",
+          "MJPEG/JPEG quality (integer 0:min 100:max)",
+          0, 100, GST_OMX_JPEG_ENC_QUALITY_DEFAULT, G_PARAM_READWRITE));
+
+  videoenc_class->set_format = GST_DEBUG_FUNCPTR (gst_omx_jpeg_enc_set_format);
+  videoenc_class->get_caps = GST_DEBUG_FUNCPTR (gst_omx_jpeg_enc_get_caps);
+  videoenc_class->cdata.default_src_template_caps = "image/jpeg, "
```

Comment:
I think we should be installing our own caps templates

Reply:
If we install a new pad template the old one will be replaced. This wasn't possible before.